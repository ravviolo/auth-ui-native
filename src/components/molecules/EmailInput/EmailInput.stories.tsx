import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { StoryScreen } from 'storybook/StoryScreen';

import { EmailInput } from './EmailInput';

storiesOf('EmailInput', module)
  .addDecorator((getStory) => <StoryScreen>{getStory()}</StoryScreen>)
  .add('Default', () => <EmailInput testID={text('testID', 'email-input-test-id')} />);
