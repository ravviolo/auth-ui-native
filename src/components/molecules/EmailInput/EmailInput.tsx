import { InputField, InputLabel, TextInputHandle } from 'components/atoms';
import { forwardRef } from 'react';
import { View } from 'react-native';

interface Props {
  testID: string;
}

export const EmailInput = forwardRef<TextInputHandle, Props>(({ testID }, ref) => {
  return (
    <View testID={testID}>
      <InputLabel label="Email" testID="input-label-email-test-id" />
      <InputField ref={ref} testID="input-field-email-test-id" />
    </View>
  );
});
