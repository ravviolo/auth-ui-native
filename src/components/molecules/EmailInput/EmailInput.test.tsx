import { fireEvent, render, screen, within } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { EmailInput } from './EmailInput';

import type { Props } from './EmailInput';

describe('EmailInput', () => {
  const props: Props = {
    testID: 'email-input-test-id',
  };

  it('should render input with text entry not secured', () => {
    render(<EmailInput {...props} />);

    const emailInput = screen.getByTestId('email-input-test-id');
    const emailInputField = within(emailInput).getByTestId('input-field-email-test-id');

    expect(emailInputField.props).toHaveProperty('secureTextEntry', false);
  });

  it('should render initially as empty input', () => {
    render(<EmailInput {...props} />);

    const emailInput = screen.getByTestId('email-input-test-id');
    const emailInputField = within(emailInput).getByTestId('input-field-email-test-id');

    expect(emailInputField.props.value).toBe('');
  });

  it('should handle onChangeText events', () => {
    render(<EmailInput {...props} />);

    const emailInput = screen.getByTestId('email-input-test-id');
    const emailInputField = within(emailInput).getByTestId('input-field-email-test-id');

    fireEvent.changeText(emailInputField, 'test input');

    expect(emailInputField.props.value).toBe('test input');
  });

  it("should display 'Email' label", () => {
    render(<EmailInput {...props} />);

    const emailInput = screen.getByTestId('email-input-test-id');
    const emailInputLabel = within(emailInput).getByTestId('input-label-email-test-id');

    expect(emailInputLabel).toHaveTextContent('Email');
  });

  it('should match snapshot', () => {
    const tree = testRenderer.create(<EmailInput {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
