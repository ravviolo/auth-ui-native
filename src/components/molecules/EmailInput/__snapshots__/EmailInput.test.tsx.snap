// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailInput should match snapshot 1`] = `
<View
  testID="email-input-test-id"
>
  <Text
    style={
      Object {
        "color": "#3D3D3D",
        "marginBottom": 3,
      }
    }
    testID="input-label-email-test-id"
  >
    Email
  </Text>
  <TextInput
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    secureTextEntry={false}
    style={
      Array [
        Object {
          "borderColor": "#9E9E9E",
          "borderRadius": 8,
          "borderWidth": 1,
          "height": 40,
          "padding": 10,
        },
        Object {
          "borderColor": "#9E9E9E",
        },
      ]
    }
    testID="input-field-email-test-id"
    value=""
  />
</View>
`;
